ethereum/client/src/lib.rs

```diff
@@ -212,54 +299,38 @@ impl Contracts {
         }
     }
 
-    pub async fn provide_content_message<
-        H: Convert<B256>,
-        U1: Convert<U256>,
-        U2: Convert<U256>,
-        U3: Convert<U256>,
-        N: Convert<Uint<256, 4>>,
-        S: Convert<B256>,
-        R: Convert<Address>,
-        P: Convert<Bytes>,
-    >(
+    pub async fn provide_content_message(
         &self,
-        block_number: U1,
-        total_leaves: U2,
-        leaf_index: U3,
-        nonce: N,
-        sender: S,
-        receiver: R,
-        payload: P,
-        proof: Vec<H>,
+        block_number: u64,
+        total_leaves: u64,
+        leaf_index: u64,
+        nonce: [u8; 32],
+        sender: [u8; 32],
+        receiver: [u8; 20],
+        payload: Vec<u8>,
+        proof: Vec<[u8; 32]>,
     ) -> Result<TxHash, Error> {
         let call = self.message_queue_instance.processMessage(
-            block_number.convert(),
-            total_leaves.convert(),
-            leaf_index.convert(),
+            U256::from(block_number),
+            U256::from(total_leaves),
+            U256::from(leaf_index),
             ContentMessage {
-                nonce: nonce.convert(),
-                sender: sender.convert(),
-                receiver: receiver.convert(),
-                data: payload.convert(),
+                nonce: B256::from(nonce),
+                sender: B256::from(sender),
+                receiver: Address::from(receiver),
+                data: Bytes::from(payload),
             },
-            proof.into_iter().map(|x| x.convert()).collect(),
+            proof.into_iter().map(|x| B256::from(x)).collect(),
         );
 
         match call.estimate_gas().await {
-            Ok(_gas_used) => {
-                match call
-                    .from(self.signer.address())
-                    .gas_price(U256::from(2_000_000_000_000u128))
-                    .send()
-                    .await
-                {
-                    Ok(pending_tx) => Ok(*pending_tx.tx_hash()),
-                    Err(e) => {
-                        log::error!("Sending error: {e:?}");
-                        Err(Error::ErrorSendingTransaction(e))
-                    }
+            Ok(_gas_used) => match call.send().await {
```

Comment:
What the gas price will be used here?

Reply:
Gas price determined by GasFiller in Alloy  Provider that is added by calling with_recommended_fillers()